src/gatsby/utils/resolvers/index.js

```diff
@@ -0,0 +1,79 @@
+const path = require('path')
+
+async function firstNodeOfType(context, children, type) {
+  return (await context.nodeModel.getNodesByIds({ ids: children, type }))[0]
+}
+
+async function descendNodeByTypes(context, children, types) {
+  const type = Array.isArray(types) && types[0]
+  const currentNode = await firstNodeOfType(context, children, type)
+  console.log(types, currentNode)
+  if (currentNode) {
+    if (types.length <= 1) {
+      return currentNode
+    } else {
+      return descendNodeByTypes(context, currentNode.children, types.slice(1))
+    }
+  } else {
+    return null
+  }
+}
+
+function resolveNodeFromRelativeFile(type) {
+  return async function (source, args, context, info) {
+    const relativeChildPath = context.defaultFieldResolver(
+      source,
+      args,
+      context,
+      info
+    )
+    const childSourcePath = path.relative('../', relativeChildPath)
+    return context.nodeModel.runQuery({
+      query: {
+        filter: {
+          sourcePath: {
+            eq: childSourcePath
+          }
+        }
+      },
+      type,
+      firstOnly: true
+    })
+  }
+}
+
+function resolveRelativeImage() {
```

Comment:
does it repeat `gatsby-remark-relative-images` that we had before? (I know that the newer version of `gatsby-remark-relative-images` failed, but should we at least remove it from the dependcies?)

Reply:
This bit does perform the same functionality for custom nodes as `remark-relative-images` does with frontmatter, but that plugin is still useful for any images in the markdown body content.

Reply:
frontmatter image URLs are already relative ... so still not sure I understand what's happening here ... can we put some explanation? how was it handled before? can we change .md files to avoid path transformations at least?

Reply:
sounds good, let's get rid of it 👍